### PR TITLE
[#5560] Add support for extra catalogs in the Gravitino CLI

### PR DESCRIPTION
**Pull request by justinmclean**
### What changes were proposed in this pull request?

Expand Gravitino CLI support.

### Why are the changes needed?

Add support for all current catalogs.

Fix: #5560

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Tested locally.

### CODE REVIEW

**Comment by tengqm** [comment]
Is this PR based on #5717?

**Comment by justinmclean** [comment]
Yes it is based on https://github.com/apache/gravitino/pull/5717

**Comment by justinmclean** [comment]
@shaofengshi / @tengqm  this is ready to review

File: docs/cli.md [hunk 1]
```diff
@@ -296,6 +296,30 @@ gcli catalog create  -name postgres --provider postgres --properties jdbc-url=jd
...
+##### Create a Paimon catalog
+
+```bash
+gcli catalog create --name paimon --provider paimon --properties catalog-backend=jdbc,uri=jdbc:mysql://127.0.0.1:3306/metastore_db,authentication.type=simple
+```
+
+#### Create a Hudi catalog
+
+```bash
+gcli catalog create --name oceanbase --provider oceanbase --properties catalog-backend=hms,uri=thrift://127.0.0.1:9083
```
Comment (shaofengshi) [inline docs/cli.md:314]: name and provider are wrong, need be fixed.

Comment (justinmclean) [reply docs/cli.md:314]: I copied this from our own documentation - so I guess that's wrong as well?

**Review by shaofengshi — APPROVED** [review]
LGTM

**Pull request merged**